pkg/adapter/sinker.go

```diff
@@ -20,13 +20,8 @@ type sinker struct {
 }
 
 func (s *sinker) processEvent(ctx context.Context, request *http.Request, payload []byte) {
-	if err := s.vcx.ParseEventType(request, s.event); err != nil {
-		s.run.Clients.Log.Errorf("failed to find event type: %v", err)
-		return
-	}
-
 	var err error
-	s.event, err = s.vcx.ParsePayload(ctx, s.run, s.event, string(payload))
+	s.event, err = s.vcx.ParsePayload(ctx, s.run, request, string(payload))
```

Comment:
I think it may be good to store the request directly in events : 

```
diff --git a/pkg/params/info/events.go b/pkg/params/info/events.go
index 788c8cb2..584d5671 100644
--- a/pkg/params/info/events.go
+++ b/pkg/params/info/events.go
@@ -1,7 +1,10 @@
 package info
 
+import "net/http"
+
 type Event struct {
        Event interface{}
+  EventRequest http.Request
 
        // EventType is what coming from the provider header, i.e:
        // GitHub -> pull_request
```

i'll probably do this later on when working on webhook validation,